Rakefile

```diff
@@ -446,3 +450,43 @@ namespace :test do
     end
   end
 end
+
+namespace :dependencies do
+  task :update do
+    target_submodule = ENV['TARGET_SUBMODULE']
+    target_version = ENV['TARGET_VERSION']
+
+    local_info = Bumpsnag.get_git_info
+    target_info = Bumpsnag.get_git_info(target_submodule)
+
+    if target_submodule.nil? || target_version.nil?
+      raise 'Submodule or version targets not provided, exiting'
+      exit(1)
+    end
+
+    pp "Updating submodule: #{target_submodule} to version: #{target_version}"
+    updated = Bumpsnag.update_submodule(target_submodule, target_version)
+
+    if updated
+      target_pr = local_info[:latest_pr] + 1
+      origin_repo = 'https://github.com/bugsnag/bugsnag-unity'
+      target_repo = target_info[:origin]
+
+      changelog_message = "Update #{target_submodule} to [#{target_version}](#{target_repo}/releases/tag/#{target_version}) [##{target_pr}](#{origin_repo}/pull/#{target_pr})"
+
+      Bumpsnag.add_changelog_entry(changelog_message, 'Dependencies')
+
+      release_branch = "bumpsnag-#{target_submodule}-#{target_version}"
+
+      commit_message = "[full ci] Update #{target_submodule} to #{target_version}"
+
+      Bumpsnag.change_branch(release_branch, true)
+      Bumpsnag.commit_changes(commit_message, [target_submodule, 'CHANGELOG.md'])
+      Bumpsnag.push_changes(release_branch)
```

Comment:
This is quite programmatic, and could be folded into Bumpsnag itself.  This would also allow us to add a bin file to invoke the functions of bumpsnag, allowing us to write scripts outside of Ruby.

Reply:
I'd certainly be keen to push some of this to Bumpsnag - particularly the changelog message, release branch name and commit message - so they are consistent across other repos. Ideally we'd just pass the submodule names down and it would do the rest?

(Not suggesting we do this before the script is released and in action though.)

Reply:
Agreed - I'll spin this into a separate ticket.

Reply:
See PLAT-11074